node/src/db/models/asset_states.rs

```diff
@@ -20,11 +20,12 @@ pub struct AssetState {
     pub expiry_date: Option<DateTime<Utc>>,
     pub superseded_by: Option<uuid::Uuid>,
     pub initial_permission_bitflag: i64,
-    pub additional_data_json: Value,
+    pub initial_data_json: Value,
```

Comment:
so `initial_data_json` is the one which takes part in asset_id hashing as I got, and `additional_data_json` is not checkpointed yet as I got?

Reply:
We discussed this via a voice call earlier so just swinging back here:

The checkpoints are the hash of the state at the current append only period. We'll be leaving the initial data json alone unless the asset is migrated or perhaps a new command in the future to roll up the state into the initial state again to reduce the size of the asset (by condensing its state). The append only rows will normally be persisted for the sake of history. Should be able to verify the state of things by checking the hash of the append only data at the time of the checkpoint against what's stored in the checkpoint.